mail_environment_office365/models/ir_mail_server.py

```diff
@@ -0,0 +1,18 @@
+# Copyright 2022 Camptocamp SA (https://www.camptocamp.com).
+# @author Anna Janiszewska <anna.janiszewska@camptocamp.com>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import models
+
+
+class IrMailServer(models.Model):
+    _inherit = "ir.mail_server"
+
+    @property
+    def _server_env_fields(self):
+        base_fields = super()._server_env_fields
+        office365_fields = {
+            "smtp_authentication": {},
```

Comment:
We should merge and forward port this instead, IMO:
- https://github.com/OCA/server-env/pull/239

The `smtp_authentication` field is not specific to office365.


----

Is there no other field needed to be set through the environment, for outlook?
Like the `microsoft_outlook_access_token` or `microsoft_outlook_refresh_token` ?

Reply:
I'm not sure @ivantodorovich 
This module is a specific one. It provides a description of the parameters required for the outlook.
If we go with forward port we will need to update the description on `mail_environment`.
Maybe it's not obvious for everyone in the community.